spec/models/user_spec.rb

```diff
@@ -16,6 +16,57 @@
     end
   end
 
+  describe '.find_first_by_auth_conditions' do
+    let!(:user) { create(:user, reset_password_token: SecureRandom.uuid) }
+    subject(:find_first_by_auth_conditions) do
+      User.find_first_by_auth_conditions(conditions)
+    end
+
+    context 'when searching by email' do
+      context 'when email exists' do
+        let(:conditions) do
+          { 'email' => user.email }
+        end
+
+        it 'returns user with same email' do
+          expect(find_first_by_auth_conditions).to eq(user)
+        end
+      end
+
+      context 'when email does not exist' do
+        let(:conditions) do
+          { 'email' => 'someemail@domain.uk' }
+        end
+
+        it 'returns nil' do
+          expect(find_first_by_auth_conditions).to be_nil
+        end
+      end
+    end
+
+    context 'when searching by reset password token' do
+      context 'when token exists' do
+        let(:conditions) do
+          { reset_password_token: user.reset_password_token }
+        end
+
+        it 'returns user with same token' do
+          expect(find_first_by_auth_conditions).to eq(user)
+        end
+      end
+
+      context 'when token does not exist' do
+        let(:conditions) do
+          { reset_password_token: "#{user.reset_password_token}C" }
```

Comment:
is the `C` supposed to be in the string here?

Reply:
or is this to make it an invalid token?

Reply:
Yes I did for invalid token but I am happy to put anything else. Wdyt?

Reply:
In the interests of clarity, it would probably be better to make that string more representative of its purpose.

Reply:
I liked the "invalid_" solution. I already pushed ♻️